api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.expressions.partitions.Partition;
+import java.util.Map;
+
+public interface SupportsPartitions extends Table {
+
+  /**
+   * List all partition names of the table
+   *
+   * @return The list of partition names
+   */
+  String[] listPartitionNames();
+
+  /**
+   * List all partitions
+   *
+   * @return The list of partitions
+   */
+  Partition[] listPartitions();
```

Comment:
Do we need listPartitionsByFilter? The number of partitions may be large.

Reply:
Good idea,  but I think it is an extended interface, we should only focus on the basic API in this PR